frontend/src/api/k8s/__tests__/clusterQueues.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { k8sListResourceItems } from '@openshift/dynamic-plugin-sdk-utils';
+import { mockClusterQueueK8sResource } from '~/__mocks__/mockClusterQueueK8sResource';
+import { ClusterQueueKind } from '~/k8sTypes';
+import { listClusterQueues } from '~/api/k8s/clusterQueues';
+
+jest.mock('@openshift/dynamic-plugin-sdk-utils', () => ({
+  k8sListResourceItems: jest.fn(),
+}));
+
+const k8sListResourceItemsMock = jest.mocked(k8sListResourceItems<ClusterQueueKind>);
+
+const clusterQueueMock = mockClusterQueueK8sResource({ name: 'test-cluster-queue' });
+
+describe('listClusterQueues', () => {
+  it('should fetch and return clusterqueues', async () => {
+    k8sListResourceItemsMock.mockResolvedValue([clusterQueueMock]);
+    const result = await listClusterQueues();
+    expect(k8sListResourceItemsMock).toHaveBeenCalledWith({
+      model: {
+        apiGroup: 'kueue.x-k8s.io',
+        apiVersion: 'v1beta1',
+        kind: 'ClusterQueue',
+        plural: 'clusterqueues',
+      },
+      queryOptions: {},
+    });
+    expect(k8sListResourceItemsMock).toHaveBeenCalledTimes(1);
+    expect(result).toStrictEqual([clusterQueueMock]);
+  });
+
+  it('should handle errors and rethrow', async () => {
+    k8sListResourceItemsMock.mockRejectedValue(new Error('error1'));
+    await expect(listClusterQueues()).rejects.toThrow('error1');
+    expect(k8sListResourceItemsMock).toHaveBeenCalledTimes(1);
+    expect(k8sListResourceItemsMock).toHaveBeenCalledWith({
+      model: {
+        apiGroup: 'kueue.x-k8s.io',
```

Comment:
I see the `model` object has been used twice. Can this be moved to a variable?

Reply:
This was based on how tests in [`frontend/src/api/k8s/__tests__/pvcs.spec.ts`](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/api/k8s/__tests__/pvcs.spec.ts#L96) are written. We could import the model variable that is used in the source perhaps, but that makes this test a bit of a tautology (of course it will be called with that model value if that model value is what we are passing)... I suppose that's also true here

Reply:
@christianvogt @pnaik1 do you guys have any opinions on the best practices here?

Reply:
This is fine